Tests/Client/Unit Tests/WebSocket/WebSocketTests.swift

```diff
@@ -0,0 +1,153 @@
+//
+//  WebSocketTests.swift
+//  StreamChatClientTests
+//
+//  Created by Alexey Bukhtin on 02/05/2020.
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import XCTest
+@testable import StreamChatClient
+
+final class WebSocketTests: XCTestCase {
+    
+    let mockProvider = WebSocketProviderMock(request: URLRequest(url: URL(string: "http://test.com")!), callbackQueue: .main)
+    let logger = ClientLogger(icon: "🦄", level: .info)
+    
+    override static func setUp() {
+        WebSocket.pingTimeInterval = 2
+    }
+    
+    override class func tearDown() {
+        WebSocket.pingTimeInterval = 25
+    }
+    
+    func test_webSocket_connectWithSubscription() {
+        // Wait for connected event from 2 subscriptions.
+        let connectedExpectation = expectation(description: "WebSocket connected")
+        connectedExpectation.expectedFulfillmentCount = 2
+        
+        // On force disconnect the WebSocket shouldn't try to reconnect.
+        let shouldntReconnectExpectation = expectation(description: "WebSocket shouldn't reconnect")
+        shouldntReconnectExpectation.isInverted = true
+        
+        let eventsHandler: Client.OnEvent = { event in
+            guard case .connectionChanged(let state) = event else {
+                return
+            }
+            
+            if case .connected = state {
+                connectedExpectation.fulfill()
+            } else if case .reconnecting = state {
+                shouldntReconnectExpectation.fulfill()
+            }
+        }
+        
+        let webSocket = WebSocket(mockProvider, options: .stayConnectedInBackground, onEvent: eventsHandler)
+        _ = webSocket.subscribe(callback: eventsHandler)
+        
+        XCTAssertFalse(webSocket.isConnected)
+        webSocket.connect()
+        wait(for: [connectedExpectation], timeout: 2)
+        XCTAssertTrue(webSocket.isConnected)
+        webSocket.disconnect(reason: "test")
+        XCTAssertFalse(webSocket.isConnected)
+        wait(for: [shouldntReconnectExpectation], timeout: 1)
```

Comment:
As mentioned above, the tests should use virtual time so we don't have to wait 10+ seconds for this test suite to finish.

Reply:
I know, but this is a specific test for another internal _timer_ that should delay reconnection request on each fail attempt. We can remove this test if you don't like it.